kv/tikv/response.go

```diff
@@ -0,0 +1,103 @@
+package tikv
+
+import "github.com/pingcap-incubator/tinykv/proto/pkg/kvrpcpb"
+
+// RespResult is a 'generic' result type for responses. It is used to return a Response/error pair over channels where
+// we can't use Go's multiple return values.
+type RespResult struct {
```

Comment:
Since this struct is exported, Could we expose methods like `prewriteResponse/getResponse` and make them return multiple values: `(*kvrpcpb.XXXResponse, error)`? (The `error` value represents if the type assertion fails)